src/compiler/gcc.rs

```diff
@@ -365,7 +368,15 @@ where
                     _ => cannot_cache!("-x"),
                 };
             }
-            Some(Arch(_)) => {}
+            Some(Arch(arch)) => {
+                match seen_arch {
+                    Some(s) if &s != arch && dont_cache_multiarch => {
```

Comment:
Is it a good idea to not cache the multiarch build all the time?

Reply:
As in flip the logic so that the env var is to enable multiarch caching?

Reply:
> As in flip the logic so that the env var is to enable multiarch caching but default to not caching?

Yes, my original idea is always not caching in this case. But flip the logic also a good idea.

Reply:
I've flipped the logic and made it so that the arch flag rewriting only happens when multiple `-arch` flags are given